lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +200,87 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private async isNexusDataSource(registry: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const endPointRe = regEx(`^(?<endPoint>.*:\\d{1,5})\\/`);
+    const { endPoint } = registry.match(endPointRe)?.groups ?? {
+      endPoint: '',
+    };
+    const nexusUrl = endPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
```

Comment:
If I understand this correctly, it looks like this results in one extra http request *per dependency*?

Reply:
it's per dependency that falls into the `getDependencyFallback`,
maybe the code should move somewhere else then? 
or should we just do caching for the nexus results in here?

Reply:
I think it is resolvable using caching. It's not like a host changes its type all the time. You could cache it forever and maybe never ever get a problem. Instead though I think we could cache it for e.g. a day

Reply:
dont we have caching inside `http.getJson()` already?
PSL
https://github.com/renovatebot/renovate/blob/244e18ab77865a908ca3309cab16e4d76851dfa3/lib/util/http/index.ts#L135-L155

`useCache` should be TRUE though, check it out

Reply:
If we use `await this.http.getJson<HttpResponse>(nexusUrl, {
        useCache: true,
      });` 
@viceice will this be cached for a day by default ?

Reply:
It's also inefficient to pass it all the way down the http stack before returning the cached value, applying body rules on the way down, parsing it on the way back again, etc